domains/auth/Auth.API/Integration.Tests/Controllers/LogoutControllerTests.cs

```diff
@@ -48,37 +54,31 @@ public async Task LogoutAsync_ShouldReturnErrorCodeUrl_WhenDiscoveryCacheFails()
         var client = factory.CreateAuthenticatedClient(user);
 
         var result = await client.GetAsync("auth/logout");
+        var resultContent = await result.Content.ReadAsStringAsync();
+        var redirectUriResponse = JsonConvert.DeserializeObject<RedirectUriResponse>(resultContent);
         Assert.NotNull(result);
-
-        var query = HttpUtility.UrlDecode(result.Headers.Location?.AbsoluteUri);
-        Assert.DoesNotContain($"{ErrorCode.QueryString}=", query);
-
-        var uri = new Uri(query!);
-        Assert.Equal(oidcOptions.FrontendRedirectUri.Host, uri.Host);
-        Assert.Equal(HttpStatusCode.TemporaryRedirect, result.StatusCode);
+        Assert.Equal(oidcOptions.FrontendRedirectUri.AbsoluteUri, redirectUriResponse?.RedirectionUri);
+        Assert.Equal(HttpStatusCode.OK, result.StatusCode);
     }
 
     [Fact]
-    public async Task LogoutAsync_ShouldNotRedirectWithHint_WhenInvokedAnonymously()
+    public async Task LogoutAsync_DecodableUserDescriptorFails_ReturnsOkResultWithRedirectionUri()
     {
         var server = WireMockServer.Start().MockConfigEndpoint().MockJwksEndpoint();
 
-        var oidcOptions = new OidcOptions()
+        var oidcOptions = new OidcOptions
         {
             AuthorityUri = new Uri($"http://localhost:{server.Port}/op"),
             FrontendRedirectUri = new Uri("https://example.com")
         };
 
         var client = factory.CreateAnonymousClient(config: builder => builder.ConfigureTestServices(services => services.AddScoped(_ => oidcOptions)));
-
         var result = await client.GetAsync("auth/logout");
-        Assert.NotNull(result);
+        var resultContent = await result.Content.ReadAsStringAsync();
+        var redirectUriResponse = JsonConvert.DeserializeObject<RedirectUriResponse>(resultContent);
 
-        var query = HttpUtility.UrlDecode(result.Headers.Location?.AbsoluteUri);
-        var uri = new Uri(query!);
-        Assert.Equal(oidcOptions.FrontendRedirectUri.Host, uri.Host);
-        Assert.Equal(HttpStatusCode.TemporaryRedirect, result.StatusCode);
-        Assert.DoesNotContain($"post_logout_redirect_uri={oidcOptions.FrontendRedirectUri.AbsoluteUri}", query);
-        Assert.DoesNotContain($"id_token_hint", query);
+        Assert.NotNull(result);
```

Comment:
This `assert` should happen before we use the variable. Also the very next thing should be the HTTP status code OK assert.